README.md

```diff
@@ -138,6 +138,70 @@ object Main extends App {
 }
 ```
 
+##### Basic Usage Streaming
+
+Assuming you have already created a `FaunaClient`.
+
+###### Flow subscriber
+
+It is possible to use the `java.util.concurrent.Flow` API directly by binding a `Subscriber` manually.
+
+In the example below we are capturing the 4 first messages.
+
+```scala
+client.stream(docRef).flatMap { publisher =>
+  // Promise to hold the final state
+  val capturedEventsP = Promise[List[Value]]
+
+  // Our manual Subscriber
+  val valueSubscriber = new Flow.Subscriber[Value] {
+    var subscription: Flow.Subscription = null
+    val captured = new ConcurrentLinkedQueue[Value]
+
+    override def onSubscribe(s: Flow.Subscription): Unit = {
+      subscription = s
+      subscription.request(1)
+    }
+
+    override def onNext(v: Value): Unit = {
+      captured.add(v)
+      if (captured.size() == 4)
+        capturedEventsP.success(captured.iterator().asScala.toList)
+      else
+        subscription.request(1)
+    }
+
+    override def onError(t: Throwable): Unit =
+      capturedEventsP.failure(t)
+
+    override def onComplete(): Unit =
+      capturedEventsP.failure(new IllegalStateException("not expecting the stream to complete"))
+  }
+  // subscribe to publisher
+  publisher.subscribe(valueSubscriber)
+  // wait for Future completion
+  capturedEventsP.future
+}
+
+```
+
+###### Monix
+
+The [reactive-streams](http://www.reactive-streams.org/) standard offers a strong interoperability in the streaming ecosystem.
+
+We can replicate the previous example using the [Monix](https://monix.io/) streaming library.
+
+```scala
+client.stream(docRef).flatMap { publisher =>
```

Comment:
Whatever direction for `docRef` that is applied in the Flow example, it should be applied here too.